### PR TITLE
Mikrotik cAP ac patchset

### PR DESCRIPTION
**Pull request by vibornoff**

### CODE REVIEW

**Comment by vibornoff** [comment]
@alaraun Any move on this PR?

File: target/linux/ipq40xx/base-files/etc/board.d/01_leds [hunk 1]
```diff
@@ -45,6 +45,14 @@ engenius,ens620ext)
 	ucidef_set_led_netdev "lan1" "LAN1" "green:lan1" "eth0"
 	ucidef_set_led_netdev "lan2" "LAN2" "green:lan2" "eth1"
 	;;
+mikrotik,cap-ac)
+	ucidef_set_led_default "power" "POWER" "blue:power" "1"
+	ucidef_set_led_default "user" "USER" "green:user" "0"
+	ucidef_set_led_netdev "eth1" "ETH1" "green:eth1" "eth1"
```
Comment (alaraun) [inline target/linux/ipq40xx/base-files/etc/board.d/01_leds]: It is confusing to have different eth names for the same eth devices in different places.

**Comment by alaraun** [comment]
I have built the image with the current changes and I was not able to get dynamic IP from lan interface after flashing.

File: target/linux/ipq40xx/base-files/etc/board.d/01_leds (hunk 1 above)
Comment (vibornoff) [reply target/linux/ipq40xx/base-files/etc/board.d/01_leds]: Take a look into a diagram:
```
  CPU       switch0
+------+   +--------+
| eth0 |---| port 0 |
| eth1 |   | port 1 |
+------+   | port 2 |                Enclousure
   |       | port 3 |     PHY          label
   |       | port 4 |---ethphy3---< E2/PoE out
   +-------| port 5 |---ethphy4---< E1/PoE in
           +--------+
```

So LEDs are named according to its enclouse labels:
* "green:eth1" for "E1/PoE in" labeled port
* "green:eth2" for "E2/PoE out" labeled port

Attaching LED triggers to switch port events didn't work for some reason. So I attached to netdev events and it worked like a charm:
* "green:eth1" to "eth1" device
* "green:eth2" to "eth0" device

Any idea on how to get it working better?

**Comment by vibornoff** [comment]
> I have built the image with the current changes and I was not able to get dynamic IP from lan interface after flashing.

Not sure but isn't it a default behavior of all fresh OpenWRT setups? If I remember it right I had to configure static IP from 192.168.1.0/24 on my laptop to get access to my old WDR4300 after flashing.

**Comment by alaraun** [comment]
> > I have built the image with the current changes and I was not able to get dynamic IP from lan interface after flashing.
> 
> Not sure but isn't it a default behavior of all fresh OpenWRT setups? If I remember it right I had to configure static IP from 192.168.1.0/24 on my laptop to get access to my old WDR4300 after flashing.

eth2 does not respond even when setting computer IP to static 192.168.1.x. It mean cAP ac is not usable by default build.

**Comment by vibornoff** [comment]
> eth2 does not respond even when setting computer IP to static 192.168.1.x. It mean cAP ac is not usable by default build.

Yep, my fault. Switch-related patch is not yet ready. I'll rearrange my commits shortly with only LEDs-related stuff.

**Comment by vibornoff** [comment]
Switch patch seems ready. VLANs work too.
However there is one minor issue. E1/E2 don't blink on switch port tx/rx for some reason, only link status events trigger those leds to lit on and off.

**Comment by vibornoff** [comment]
Hello! Hope you have ever chance to take a look to this PR. It is absolutely viable to be merged, especially switch-related patch.
BTW, my awesome WDR4300 didn't survive a power surge so I had to flash cAP-ac, and it just works.

**Pull request closed without merging**